File: src/main/java/buttondevteam/discordplugin/DiscordPlayer.java
package buttondevteam.discordplugin;

import buttondevteam.discordplugin.mcchat.MCChatPrivate;
import buttondevteam.lib.player.ChromaGamerBase;
import buttondevteam.lib.player.UserClass;

@UserClass(foldername = "discord")
public class DiscordPlayer extends ChromaGamerBase {
	private String did;
	// private @Getter @Setter boolean minecraftChatEnabled;

	public DiscordPlayer() {
	}

	public String getDiscordID() {
		if (did == null)
			did = plugindata.getString(getFolder() + "_id");
		return did;
	}

	/**
	 * Returns true if player has the private Minecraft chat enabled. For setting the value, see
	 * {@link MCChatPrivate#privateMCChat(sx.blah.discord.handle.obj.MessageChannel, boolean, sx.blah.discord.handle.obj.IUser, DiscordPlayer)}
	 */
	public boolean isMinecraftChatEnabled() {
		return MCChatPrivate.isMinecraftChatEnabled(this);
	}
}

File: src/main/java/buttondevteam/discordplugin/DiscordPlayer.scala
package buttondevteam.discordplugin

import buttondevteam.discordplugin.mcchat.MCChatPrivate
import buttondevteam.lib.player.{ChromaGamerBase, UserClass}

@UserClass(foldername = "discord") class DiscordPlayer() extends ChromaGamerBase {
    private var did: String = null

    // private @Getter @Setter boolean minecraftChatEnabled;
    def getDiscordID: String = {
        if (did == null) did = getFileName
        did
    }

    /**
     * Returns true if player has the private Minecraft chat enabled. For setting the value, see
     * {@link MCChatPrivate# privateMCChat ( MessageChannel, boolean, User, DiscordPlayer)}
     */
    def isMinecraftChatEnabled: Boolean = MCChatPrivate.isMinecraftChatEnabled(this)
}